Core/Analyzers/ClassAnalyzer.cpp

```diff
@@ -67,21 +102,35 @@ void ClassAnalyzer::run()
     for (auto address = sectionStart; address < sectionEnd; address += 8) {
         ClassInfo ci;
         ci.listPointer = address;
-        ci.address = arp(m_file->readLong(address));
-        ci.dataAddress = arp(m_file->readLong(ci.address + 0x20));
+        ci.address = arp(m_file->readPointer(address));
+
+		int pointerSize = m_file->pointerSize();
+
+		// dataAddress is class_ro
+        ci.dataAddress = arp(m_file->readPointer(ci.address + (4 * pointerSize)));
 
         // Sometimes the lower two bits of the data address are used as flags
         // for Swift/Objective-C classes. They should be ignored, unless you
         // want incorrect analysis...
         ci.dataAddress &= ~ABI::FastPointerDataMask;
 
-        ci.nameAddress = arp(m_file->readLong(ci.dataAddress + 0x18));
+		// kat: core desperately needs a 'load pre-defined struct from addr' ;_;
+		int RO_PTR_START = pointerSize == 8 ? 16 : 12;
+		int RO_NAME_OFF_FROM_PTR_START = RO_PTR_START + pointerSize * 1;
+		int RO_METH_OFF_FROM_PTR_START = RO_PTR_START + pointerSize * 2;
+		int RO_IVAR_OFF_FROM_PTR_START = RO_PTR_START + pointerSize * 4;
```

Comment:
These should follow the normal naming convention, e.g. `roPointerStart`, `roMethodOffset`, etc, etc.